src/call.c

```diff
@@ -2322,12 +2337,12 @@ static void sipsess_progr_handler(const struct sip_msg *msg, void *arg)
 	}
 
 	if (media) {
-		call_stream_start(call, false);
 		call_event_handler(call, CALL_EVENT_PROGRESS, call->peer_uri);
+		call_stream_start(call, false);
 	}
 	else {
-		call_stream_stop(call);
 		call_event_handler(call, CALL_EVENT_RINGING, call->peer_uri);
+		call_stream_stop(call);
```

Comment:
Not sure about this.

The handler may deref the call object.
When returning it will operate on a invalid call object.

Reply:
Right, I have to analyze old commits. We had this problem once. There might be a general problem if an event handler is allowed to deref the call. Especially, if there are multiple event handlers.

Reply:
Would it be a good idea to:
```c
	mem_ref(call);
	if (media) {
		call_event_handler(call, CALL_EVENT_PROGRESS, call->peer_uri);
		call_stream_start(call, false);
		mem_deref(call);
	}
	else {
		call_event_handler(call, CALL_EVENT_RINGING, call->peer_uri);
		call_stream_stop(call);
		mem_deref(call);
	}
```

Reply:
This https://github.com/baresip/baresip/pull/2403/commits/16a28ce26b77a4d39d1dd5caf66f145996f8cf0a ensures that the `call_stream_start()` does not operate on a destroyed call object.